storagemarket/impl/provider.go

```diff
@@ -595,22 +594,30 @@ func (p *Provider) start(ctx context.Context) error {
 	err := p.migrateDeals(ctx)
 	publishErr := p.readyMgr.FireReady(err)
 	if publishErr != nil {
-		log.Warnf("Publish storage provider ready event: %s", err.Error())
+		log.Warnf("publish storage provider ready event: %s", err.Error())
 	}
 	if err != nil {
-		return fmt.Errorf("Migrating storage provider state machines: %w", err)
+		return fmt.Errorf("migrating storage provider state machines: %w", err)
 	}
 
 	var deals []storagemarket.MinerDeal
 	err = p.deals.List(&deals)
 	if err != nil {
 		return xerrors.Errorf("failed to fetch deals during startup: %w", err)
 	}
+
+	// re-track all deals for whom we still have a local blockstore.
+	for _, d := range deals {
+		if _, err := os.Stat(d.InboundCAR); err == nil && d.Ref != nil {
+			_, _ = p.stores.GetOrOpen(d.ProposalCid.String(), d.InboundCAR, d.Ref.Root)
+		}
+	}
```

Comment:
We now load all in progress deals whose inbound CAR still exists on disk into the trackers. This reduces complexity of the former FinalizeReadWriteBlockstore on providerDealEnvironment, which used to accept the root CID and the path to reload the entry in the tracker if it didn't exist. It puzzled me quite a bit. I believe restoring state on startup is more correct.